addons/product/models/product_pricelist.py

```diff
@@ -55,42 +55,7 @@ def name_get(self):
 
     @api.model
     def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_uid=None):
-        if name and operator == '=' and not args:
-            # search on the name of the pricelist and its currency, opposite of name_get(),
-            # Used by the magic context filter in the product search view.
-            query_args = {'name': name, 'limit': limit, 'lang': get_lang(self.env).code}
-            query = """SELECT p.id
-                       FROM ((
-                                SELECT pr.id, pr.name
-                                FROM product_pricelist pr JOIN
-                                     res_currency cur ON
-                                         (pr.currency_id = cur.id)
-                                WHERE pr.name || ' (' || cur.name || ')' = %(name)s
-                            )
-                            UNION (
-                                SELECT tr.res_id as id, tr.value as name
-                                FROM ir_translation tr JOIN
-                                     product_pricelist pr ON (
-                                        pr.id = tr.res_id AND
-                                        tr.type = 'model' AND
-                                        tr.name = 'product.pricelist,name' AND
-                                        tr.lang = %(lang)s
-                                     ) JOIN
-                                     res_currency cur ON
-                                         (pr.currency_id = cur.id)
-                                WHERE tr.value || ' (' || cur.name || ')' = %(name)s
-                            )
-                        ) p
-                       ORDER BY p.name"""
-            if limit:
-                query += " LIMIT %(limit)s"
-            self._cr.execute(query, query_args)
-            ids = [r[0] for r in self._cr.fetchall()]
-            # regular search() to apply ACLs - may limit results below limit in some cases
-            pricelist_ids = self._search([('id', 'in', ids)], limit=limit, access_rights_uid=name_get_uid)
-            if pricelist_ids:
-                return pricelist_ids
-        return super()._name_search(name, args, operator=operator, limit=limit, name_get_uid=name_get_uid)
+        return self._name_search_fields(name, ['name', 'currency_id'], args, operator, limit, name_get_uid)
```

Comment:
From what I understood, the former behavior is only trying to add the translations, which is the default for translatable strings AFAIK.
But I will need to test indeed

Reply:
Ok, I misunderstood the function.

However, this code seems outdated.
It was introduced here https://github.com/odoo/odoo/commit/6de0051315daaeb6e49a9caa151bce338738187b
And was used in `product.product._product_price` using the context key `pricelist`, which is now only used here
https://github.com/odoo/odoo/blob/1d85cabb50cba5e330707c4041e03eaa7f27bbe4/addons/website_sale/models/product_template.py#L229
https://github.com/odoo/odoo/blob/1d85cabb50cba5e330707c4041e03eaa7f27bbe4/addons/product/models/product_template.py#L1291
https://github.com/odoo/odoo/blob/2712bf589165df7767ef036d3fe3174787ebcb11/addons/sale/views/sale_order_views.xml#L397
https://github.com/odoo/enterprise/blob/d61e33563e34bfdddc0234e4a805c901a40660ab/industry_fsm_sale/models/project_task.py#L304
https://github.com/odoo/enterprise/blob/1de608cafe62f921bf5f9a0da82a76f2eb510759/sale_renting/views/rental_views.xml#L101

Everywhere it is used, it is either a recorset or an id, and not a string like expected by 6de0051315daaeb6e49a9caa151bce338738187b

I didn't try to blame a lot further than this, but the only changes in that function since it was added are ORM/generic changes (excluding https://github.com/odoo/odoo/commit/d8d312768cc59fcde30daf34401e831092f15d39 which was done not long after the function was added):
https://github.com/odoo/odoo/commit/1aa5bc0fdfa402ac7cccbc3a7203ec5050f6339b
https://github.com/odoo/odoo/commit/30bb41d588b940e264be462355ae2872cf095c6b
https://github.com/odoo/odoo/commit/473c74f6368b9ce4c2242c53263811a925837855


______

My point being: this seems like an unused function and maybe we should remove it completely?
I would love to link the coverage build to see if the test suite uses that function, but the build isn't built anymore.